### PR TITLE
Azure: archive storage accounts used for reference images

### PR DESCRIPTION
**Pull request by mboersma**
What this PR does / why we need it:

Azure has a limit on the number of storage accounts in use in a region, and we have sometimes hit that limit while building reference images for CAPZ. This PR enhances our periodic cleanup script to archive "keeper" build artifacts into a single archive storage account. This will keep us from hitting the limit as we continue to add reference images for new Kubernetes versions. It also specifies the "Cool" storage tier which should save some $$ since these files are very rarely accessed.

Which issue(s) this PR fixes:

N/A

**Additional context**

I've tested this and run it to completion several times now to clean up our actual storage account, so it's basically sound and working.

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
I was looking at the resulting blobs in the sponsorship sub and noticed that the disks had access tier "n/a" whereas the json file said "cool" tier in the archive storage account. I did some digging and found: https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-pageblob-overview?tabs=dotnet#restrictions which is unfortunate. It seems that we may not be able to use Cool tier for the VHD files themselves.

**Comment by mboersma** [comment]
> we may not be able to use Cool tier for the VHD files themselves.

Yes, AFAICT that's the case since they're Page Blobs. At least we are using fewer separate storage accounts now.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Comment by mboersma** [comment]
@CecileRobertMichon @jsturtevant I added documentation to the script so that other MS maintainers could run it. 

I re-ran it yesterday as a sanity check, and it deleted 14 unreferenced storage accounts (failed builds) and archived 12 more.

**Comment by CecileRobertMichon** [comment]
@mboersma thoughts on running this in a daily pipeline so we don't have to remember to run it manually every once in a while?

File: images/capi/packer/azure/scripts/delete-unused-storage.sh [hunk 1]
```diff
@@ -14,9 +14,25 @@
...
+# building CAPZ reference images. It also archives existing accounts into one
+# main storage account to reduce the limited number of accounts in use.
+# Usage:
+#  <DRYRUN=true|false> delete-unused-storage.sh
+#
+# The script expects to find the `pub` tool (https://github.com/devigned/pub) and
+# the `az` CLI tool (https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+# in the PATH.
+#
+# In order to run this script, log in to the "Microsoft Azure Sponsorship" account
```
[the remark this answers is not in the document]
Comment (mboersma) [reply images/capi/packer/azure/scripts/delete-unused-storage.sh]: Good catch, I changed it to "log in to the publishing account."

**Comment by mboersma** [comment]
> thoughts on running this in a daily pipeline...

Yes, we should, now that this has been tested and seems reliable. I can try to add the job to this PR, or do it in a separate one.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**